torchx/c_src/torchx.cpp

```diff
@@ -134,12 +208,10 @@ create_tensor_resource(ErlNifEnv *env, torch::Tensor tensor)
 
 NIF(delete_tensor)
 {
-  TENSOR_PARAM(0, t);
-
-  t->~Tensor();
-  enif_release_resource(t);
+  TensorP tensor(env, argv[0]);
 
-  return nx::nif::ok(env);
+  return tensor.deallocate()? nx::nif::ok(env):
+    enif_make_badarg(env);
```

Comment:
Maybe instead of raising we should return `:already_deallocated`?

Reply:
I prefer elixir code keep the API interface. I don't prefer C++ code has special atom here.